### PR TITLE
[flutter_maps_firestore] Migrated to null-safety 

### LINKED ISSUE #725: [flutter_maps_firestore] Migrate to null-safety
Hey @AyushBherwani1998  @RedBrogdon  as recently  [google_maps_flutter](https://pub.dev/packages/google_maps_flutter/versions/2.0.0-nullsafety) package migrated to null safety. I think we can go for migrating  `flutter_maps_firestore` sample to support null safety.

/Cc @domesticmouse  @filiph

### PR DESCRIPTION
**Pull request by adarsh-technocrat**
- Upgraded dependencies
- Migrated sample to null-safety

closes #725 

/Cc @filiph @domesticmouse @RedBrogdon @AyushBherwani1998

### CODE REVIEW

File: flutter_maps_firestore/lib/main.dart [hunk 1]
```diff
@@ -49,7 +49,7 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  Stream<QuerySnapshot> _iceCreamStores;
+  Stream<QuerySnapshot>? _iceCreamStores;
```
Comment (RedBrogdon) [inline flutter_maps_firestore/lib/main.dart:52]: Rather than nullable, just mark this late, since it's initialized in `initState`.

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
On my machine, I'm getting an error that the new version of the location plugin requires that the app be migrated to the new Android plugin embedding. 

Can you take a look at the guide and run through the steps for the "Full app migration?"

https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects#full-flutter-app-migration

File: flutter_maps_firestore/lib/main.dart (hunk 1 above)
Comment (adarsh-technocrat) [reply flutter_maps_firestore/lib/main.dart:52]: Sure  @RedBrogdon.

**Comment by adarsh-technocrat** [comment]
> On my machine, I'm getting an error that the new version of the location plugin requires that the app be migrated to the new Android plugin embedding.
> 
> Can you take a look at the guide and run through the steps for the "Full app migration?"
> 
> https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects#full-flutter-app-migration

Sure @RedBrogdon  I'll go through the steps for "Full App Migration". I have not tested the sample because I don't have an IOS device. I'll go and update the PR :)

**Comment by adarsh-technocrat** [comment]
> On my machine, I'm getting an error that the new version of the location plugin requires that the app be migrated to the new Android plugin embedding.
> 
> Can you take a look at the guide and run through the steps for the "Full app migration?"
> 
> https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects#full-flutter-app-migration

But @RedBrogdon  the sample  `flutter_maps_firestore` does not contain the `android` folder!

**Comment by RedBrogdon** [comment]
> But @RedBrogdon the sample `flutter_maps_firestore` does not contain the `android` folder should I go for `flutter create .` command to make the sample run on android?

No, I'm sorry -- I've led you astray. Please just remove the /android subfolder entirely. I think you're good to land this once that's done, though I'll need to find a way to test it.

**Comment by adarsh-technocrat** [comment]
> > But @RedBrogdon the sample `flutter_maps_firestore` does not contain the `android` folder should I go for `flutter create .` command to make the sample run on android?
> 
> No, I'm sorry -- I've led you astray. Please just remove the /android subfolder entirely. I think you're good to land this once that's done, though I'll need to find a way to test it.

Done @RedBrogdon I have kept the sample as it was previous :)

**Pull request merged**